src/test/regress/input/temp_tablespaces.source

```diff
@@ -37,6 +37,57 @@ create tablespace mytempsp2 location '@testtablespace@_mytempsp2';
 create tablespace mytempsp3 location '@testtablespace@_mytempsp3';
 create tablespace mytempsp4 location '@testtablespace@_mytempsp4';
 
+--start_ignore
+CREATE EXTENSION IF NOT EXISTS plpython3u;
+--end_ignore
+CREATE OR REPLACE FUNCTION gp_tablespace_watch_start(dbid int, name text, location text)
+    RETURNS SETOF void
+    EXECUTE ON ALL SEGMENTS
+AS $$
+import subprocess
+output = '/tmp/gp_temp_tablespaces_dbid%d_%s' % (dbid, name)
+cmd = "/tmp/file_monitor %s | tee -a /tmp/gp_temp_tablespaces_dbid%d.log > %s &" % (location, dbid, output)
+cmd = ["bash", "-c", cmd]
+process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+rc = process.wait()
+if rc != 0:
+    output, error = process.communicate()
+    raise Exception("error=%s" % str(error))
+return []
```

Comment:
Can you do this and change the return type to just `void` Same with `..watch_stop()`?
```suggestion
return
```

Reply:
GPDB 7 outputs
```
ERROR:  EXECUTE ON ALL SEGMENTS is only supported for set-returning functions
```
Also, all `EXECUTE ON ALL SEGMENTS` in GPDB 7 tests seem to use `SETOF`, even for `void`. Are there other solutions for this?

Reply:
I looked into it previously, and I guess not. For GPDB 7, there's now an explicit check that prohibits the creation of non-set-returning functions with EXECUTE ON clause.